py/dml/provisional.py

```diff
@@ -0,0 +1,65 @@
+# © 2024 Intel Corporation
+# SPDX-License-Identifier: MPL-2.0
+
+import abc
+from . import logging
+from . import messages
+
+class ProvisionalFeature(abc.ABC):
+    def tag(self) -> str:
+        return self.__class__.__name__
+
+    @abc.abstractproperty
+    def __doc__(self): pass
+
+    @abc.abstractproperty
+    def short(self) -> str: pass
+
+
+# tag -> feature
+features: dict[str, ProvisionalFeature] = {}
+
+
+def feature(cls: type[ProvisionalFeature]):
+    assert issubclass(cls, ProvisionalFeature)
+    singleton = cls()
+    features[singleton.tag()] = singleton
+    return singleton
+
+
+@feature
+class walrus_params(ProvisionalFeature):
```

Comment:
I'm not fond of the name! Walrus params to me was an internal naming scheme.
Not that I'm not sure of a better name... maybe `explicit_param_overrides`? Or, if it's not too long, `explicit_param_decls_and_overrides`.

Reply:
What am I thinking, the best would be `explicit_param_decls`.

Reply:
changed!